lib/ansible/modules/storage/zfs/zpool.py

```diff
@@ -0,0 +1,454 @@
+    return tail or ntpath.basename(head)
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            state=dict(type='str', required=True, choices=['absent', 'present']),
+            raid_level=dict(type='str', required=False, choices=['raid0', 'mirror', 'raidz', 'raidz1', 'raidz2', 'raidz3']),
+            vdev=dict(type='int', require=False),
+            devices=dict(type='list', default=None),
+            spare=dict(type='list', default=None),
+            add=dict(type='bool', default=False),
+            ashift=dict(type='int', default=0, choices=[0, 9, 10, 11, 12, 13, 14, 15, 16]),
+            sets=dict(type='bool', default=False),
+            autoreplace=dict(type='bool'),
+            autoexpand=dict(type='bool'),
+            zil=dict(type='str', default=None),
+            l2arc=dict(type='str', default=None),
+        ),
+        supports_check_mode=True,
+        required_together=[['raid_level', 'devices'], ['devices', 'vdev']],
+        mutually_exclusive=[['add', 'sets']]
+    )
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+    add = module.params.get('add')
+    raid_level = module.params.get('raid_level')
+    devices = module.params.get('devices')
+    spare = module.params.get('spare')
+    vdev = module.params.get('vdev')
+    ashift = module.params.get('ashift')
+    sets = module.params.get('sets')
+    autoreplace = module.params.get('autoreplace')
+    autoexpand = module.params.get('autoexpand')
+    l2arc = module.params.get('l2arc')
+    zil = module.params.get('zil')
+
+    if autoexpand is True and (sets is True or add is True):
+        autoexpand = "autoexpand=on"
+    elif autoexpand is True and sets is False:
+        autoexpand = "-o autoexpand=on"
+    elif autoexpand is False and sets is True:
+        autoexpand = "autoexpand=off"
+    else:
+        autoexpand = ""
+
+    if autoreplace is True and (sets is True or add is True):
+        autoreplace = "autoreplace=on"
+    elif autoreplace is True and sets is False:
+        autoreplace = "-o autoreplace=on"
+    elif autoreplace is False and sets is True:
+        autoreplace = "autoreplace=off"
+    else:
+        autoreplace = ""
+
+    if raid_level is None or 'raid0' in raid_level:
+        raid_level = ''
+
+    if zil:
+        zil = ' log ' + zil
+    else:
+        zil = ''
+
+    if l2arc:
+        l2arc = ' cache ' + l2arc
+    else:
+        l2arc = ''
+
+    if ashift is None:
+        ashift = 0
+
+    if devices is None:
+        devices = ''
+    else:
+        if vdev > 1:
+            device = ''
+            for i in range(0, len(devices), vdev):
+                temp = ' ' + raid_level + ' ' + ' '.join(devices[i:i + vdev])
+                device += temp
+            devices = device
+    if not spare:
+        spare = ''
+    else:
+        spare = 'spare ' + ' '.join(spare)
+
+    result = dict(
+        name=name,
```

Comment:
We tend to only add information to the return information that is new and coming from the system.
There is no added value to return the existing information back to the user, the user already knows this as it supplied this to the module himself.